### DISCUSSION TITLE
Having trouble pointing gcc compiler to header file, Have use -l, -L, & -I. I am pretty Stuck and need much Help. Thanks

### DISCUSSION THREAD

**Discussion by 1NEGROUP** [raysan5/raylib#2880] [Q&A]
Having trouble pointing gcc compiler to header file, Have use -l, -L, & -I. I am pretty Stuck and need much Help. Thanks
[r/cpp_questions](https://www.reddit.com/r/cpp_questions/)

I have been trying to compile my project for a while and realized my header file wasnt linked properly but am still not sure how to do so. i used g++ file.cpp and i did g++ -c file.cpp, which got me a .o file, unsure of what that is. The header file is in the same location as the cpp file, so i am not sure what else to do.
[Here is my Code](https://paste.myst.rs/96fa3fdl)

[This is me trying to compile from the Terminal](https://paste.myst.rs/zm8d5oj0)

Edit I realized i was using -l wrong. You need to g++ file.cpp  -l (lowercase L) raylib

**Comment by Grave18** [comment]
Example of make output on linux "gcc -o core/core_basic_window core/core_basic_window.c -Wall -std=c99 -D_DEFAULT_SOURCE -Wno-missing-braces -Wunused-result -s -O1 -D_DEFAULT_SOURCE -I. -I/home/grave18/dev/raylib/src -I/home/grave18/dev/raylib/src/external -I/usr/local/include -L. -L/home/grave18/dev/raylib/src -L/home/grave18/dev/raylib/src -L/usr/local/lib -lraylib -lGL -lm -lpthread -ldl -lrt -lX11 -DPLATFORM_DESKTOP"
